examples/src/main/java/com/mapbox/navigation/examples/core/RouteAlertsActivity.kt

```diff
@@ -0,0 +1,187 @@
+            // R.raw.mock_response_all_passive_maneuvers_polyline6
+        )
+            .bufferedReader()
+            .use { it.readText() }
+        DirectionsResponse.fromJson(directionsResponseJson).routes()[0]
+    }
+    private val routeLineString: LineString by lazy {
+        LineString.fromPolyline(directionsRoute.geometry()!!, Constants.PRECISION_6)
+    }
+    private val tunnelFeatures = mutableListOf<Feature>()
+    private val tunnelSource = GeoJsonSource("tunnel_source")
+    private val tunnelLayer = LineLayer("tunnel_layer", "tunnel_source")
+        .withProperties(
+            PropertyFactory.lineColor(Color.DKGRAY),
+            PropertyFactory.lineWidth(15f)
+        )
+
+    private val mapboxReplayer: MapboxReplayer by lazy {
+        MapboxReplayer().apply {
+            val replayEvents = ReplayRouteMapper().mapGeometry(directionsRoute.geometry()!!)
+            pushEvents(replayEvents)
+            seekTo(replayEvents.first())
+        }
+    }
+
+    private var navigationMapboxMap: NavigationMapboxMap? = null
+    private val mapboxNavigation: MapboxNavigation by lazy {
+        val mapboxNavigationOptions = MapboxNavigation
+            .defaultNavigationOptionsBuilder(this, Utils.getMapboxAccessToken(this))
+            .locationEngine(ReplayLocationEngine(mapboxReplayer))
+            .build()
+
+        MapboxNavigation(mapboxNavigationOptions)
+    }
+
+    @SuppressLint("MissingPermission")
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_route_alerts)
+        mapView.onCreate(savedInstanceState)
+
+        mapView.getMapAsync { mapboxMap ->
+            mapboxMap.moveCamera(CameraUpdateFactory.zoomTo(16.0))
+            mapboxMap.moveCamera(CameraUpdateFactory.tiltTo(45.0))
+            mapboxMap.setStyle(getString(R.string.mapbox_navigation_guidance_day)) { style ->
+                style.addSource(tunnelSource)
+                style.addLayer(tunnelLayer)
+                navigationMapboxMap = NavigationMapboxMap(mapView, mapboxMap, this)
+                mapboxNavigation.setRoutes(listOf(directionsRoute))
+                navigationMapboxMap?.apply {
+                    updateLocationLayerRenderMode(RenderMode.GPS)
+                    updateCameraTrackingMode(NavigationCamera.NAVIGATION_TRACKING_MODE_GPS)
+                    addProgressChangeListener(mapboxNavigation)
+                    drawRoute(directionsRoute)
+                    startCamera(directionsRoute)
+                    mapboxReplayer.play()
+                }
+            }
+        }
+        mapboxNavigation.toggleHistory(true)
+        mapboxNavigation.startTripSession()
+        mapboxNavigation.registerRouteProgressObserver(
+            object : RouteProgressObserver {
+                @SuppressLint("SetTextI18n")
+                override fun onRouteProgressChanged(routeProgress: RouteProgress) {
```

Comment:
nit: add comment explaining this part of the example

// Use route progress observers to check upcoming alerts, in this example 
// we show a message that there is a tunnel approaching
//